### ISSUE TITLE
PDM fails to install dependencies due to cachecontrol not compatible with request 2.30

### ISSUE THREAD

**Issue by Roger-luo** [pdm-project/pdm#1879]
It works fine until this morning and start failing on github actions. I've made a simple repo to reproduce the error, and you can see the detailed printing from the log here https://github.com/Roger-luo/pdm-test/actions/runs/4874743432/jobs/8696058702

any idea why it's failing?

**Comment by lambda** [comment]
I had the same failure. It was due to an [update of `requests`](https://github.com/psf/requests/releases/tag/v2.30.0) that pulled in [an updated version of `urllib3`](https://urllib3.readthedocs.io/en/latest/v2-migration-guide.html) that has some breaking changes that seem to be incompatble with `cachecontrol`.

I fixed it in my environment by pinning to `requests==2.29.0`.

I haven't actually gotten to the bottom of the issue yet, but wanted to report that pinning to the older version of requests fixed the issue in my environment.
